src/test/java/uk/gov/justice/digital/service/DataStorageServiceTest.java

```diff
@@ -64,6 +64,54 @@ void shouldReturnFalseWhenStorageDoesNotExists() {
         assertFalse(actualResult);
     }
 
+    @Test
+    void shouldReturnTrueForHasRecordsWhenStorageExistsAndRecordsArePresent() throws DataStorageException {
+        val df = spark.sql("select cast(10 as LONG) as numFiles");
+
+        TableIdentifier info = new TableIdentifier("s3://test-bucket", "domain",
+                "incident", "demographics");
+        String identifier = testStorage.getTablePath(info.getBasePath(), info.getSchema(), info.getTable());
+        when(DeltaTable.isDeltaTable(spark, identifier)).thenReturn(true);
+        boolean actualResult = testStorage.exists(spark, info);
+        assertTrue(actualResult);
+
+        when(DeltaTable.forPath(spark, identifier)).thenReturn(mockedDeltaTable);
+        when(mockedDeltaTable.toDF()).thenReturn(df);
+        boolean hasRecords = testStorage.hasRecords(spark, info);
```

Comment:
minor - I'd probably inline this so you have `assertTrue(testStorage.....)` but NABD.